src/convenience/frameworks.node.ts

```diff
@@ -67,13 +66,23 @@ const azure = (context: any, req: any) => ({
     },
 });
 
+/** Vercel/Next.js Serverless Functions */
+const vercel = (req: any, res: any) => ({
+    update: Promise.resolve(req.body),
+    header: req.headers[SECRET_HEADER] as string,
+    end: () => res.end(),
+    respond: (json: string) => res.status(200).json(json),
+    unauthorized: () => res.status(401).send("secret token is wrong"),
+});
+
 // please open a PR if you want to add another
 export const adapters = {
     http,
     https: http,
     worktop,
     "aws-lambda": awsLambda,
     azure,
+    vercel,
```

Comment:
This name is misleading. Not all bots hosted on Vercel should use this adapter.

Is it possible to use this adapter when using Next.js outside of Vercel?

Reply:
Yes, it's possible. So should i name it `next.js` or `next` or `nextjs`?

Reply:
Either `nextjs` or `next.js` or `next-js` is fine by me. Just `next` would be confusing. Pick whichever one you prefer.